packages/eslint-plugin/docs/rules/no-unnecessary-condition.md

```diff
@@ -83,6 +83,18 @@ for (; true; ) {}
 do {} while (true);
 ```
 
+### `allowAssertTypePredicateParameterConditions`
```

Comment:
A lot of developers don't know what type predicates are. Could you add a line or two explaining them please?

Reply:
Yeah, definitely!